napari/_qt/qt_dims_slider.py

```diff
@@ -0,0 +1,471 @@
+
+        # having this here makes sure that using the QtDims.play() API
+        # keeps the play preferences synchronized with the play_button.popup
+        self._update_play_settings(fps, loop_mode, frame_range)
+
+        # setting fps to 0 just stops the animation
+        if fps == 0:
+            return
+
+        worker, thread = new_worker_qthread(
+            AnimationWorker,
+            self,
+            start=True,
+            connections={'frame_requested': self.qt_dims._set_frame},
+        )
+        worker.finished.connect(self.qt_dims.stop)
+        thread.finished.connect(self.play_stopped.emit)
+        self.play_started.emit()
+        self.thread = thread
+        return worker, thread
+
+
+class QtCustomDoubleSpinBox(QDoubleSpinBox):
+    """ Custom Spinbox that emits an editingFinished event both on the regular
+    editing event, as well as on the valueChanged event IF the left mouse
+    button is down.  On the standard class, the valueChanged event is too
+    spammy for the lineEdit, but the editingFinished does not pay attention
+    to the spinbox buttons.  This allows both events to be listened to."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, *kwargs)
+        self.valueChanged.connect(self.custom_change_event)
+
+    def custom_change_event(self, value):
+        if QApplication.mouseButtons() & Qt.LeftButton:
+            self.editingFinished.emit()
+
+    def textFromValue(self, value):
+        """This removes the decimal places if the float is an integer"""
+        if value.is_integer():
+            value = int(value)
+        return str(value)
+
+
+class QtPlayButton(QPushButton):
+    """Play button, included in the DimSliderWidget, to control playback
+
+    the button also owns the QtModalPopup that controls the playback settings.
+    """
+
+    play_requested = Signal(int)  # axis, fps
+
+    def __init__(self, dims, axis, reverse=False, fps=10, mode=1):
+        super().__init__()
+        self.dims = dims
+        self.axis = axis
+        self.reverse = reverse
+        self.fps = fps
+        self.mode = mode
+        self.setProperty('reverse', str(reverse))  # for styling
+        self.setProperty('playing', 'False')  # for styling
+
+        # build popup modal form
+
+        self.popup = QtModalPopup(self)
+        fpsspin = QtCustomDoubleSpinBox(self.popup)
+        fpsspin.setAlignment(Qt.AlignCenter)
+        fpsspin.setValue(self.fps)
+        fpsspin.setStepType(QDoubleSpinBox.AdaptiveDecimalStepType)
+        fpsspin.setMaximum(500)
+        fpsspin.setMinimum(0)
+
+        revcheck = QCheckBox(self.popup)
+        layout = QHBoxLayout()
+        layout.addWidget(fpsspin)
+        layout.addWidget(revcheck)
+        self.popup.form_layout.insertRow(
+            0, QLabel('frames per sec:', parent=self.popup), layout
+        )
+        self.fpsspin = fpsspin
+        self.reverse_check = revcheck
+
+        # dimsrange = dims.dims.range[axis]
```

Comment:
I take it this is for the frame range stuff, fine to wait on that for another PR and fine to leave this in here commented, but can you also add another comment so we remember what it is and why it is commented